### ISSUE TITLE
Multiple watches for changing ngLink parameters

### ISSUE THREAD

**Issue by olee** [angular/angular#7732]
I'm submitting a bug report for ngLink directive

* What is the current behavior?
In the [current ngLink directive](https://github.com/angular/angular/blob/69c1405196ab9568cc2847a0aebf342e5614fe0e/modules/angular1_router/src/ng_outlet.ts#L216-L223), each time the ngLink attribute is updated with a new route, a new $watch is started. This is because the $watch is started in the `getLink(params)` function.
Instead the $watch should be started only once at the end of the link function.

* What should happen instead?
Lines 216-223 should be moved to line 238 to prevent multiple watches from being started.

I did not test this myself, but upon reading the source code, I noticed this.
If I am wrong, I apologize for the wrong bug report.

**Comment by btford** [comment]
This is for the 1.x bits. Adding @petebacondarwin, who is working on the 1.x bits.

**Comment by mhevery** [comment]
won't fix
